blog/2023-01-17-mongodb-monitoring.md

```diff
@@ -0,0 +1,297 @@
+
+Netdata has a public [demo space](https://app.netdata.cloud/spaces/netdata-demo) (no login required) where you can explore different monitoring use-cases and get a feel for Netdata.
+
+## What MongoDB metrics are important to monitor?
+
+### Operations
+
+ - **Operations by type**: The total number of operations per type performed by the MongoDB server per second. 
+ 
+Monitoring operations by type will help to identify the types of operations that are causing the most strain on the system and can help to identify any potential areas of optimization. Additionally, it can help to ensure that the system is not overburdened and that the resources are being allocated appropriately. Monitoring this metric will also help to identify any bottlenecks or unexpected spikes in the workload, which can prevent unplanned downtime or performance degradation.
+ 
+![image](https://user-images.githubusercontent.com/24860547/212875959-01f4a957-03e0-4d1c-bc42-66062755e3ca.png)
+
+ - **Operations Lantecy**: The latencies of reads, writes and command operations performed by the MongoDB server. High latencies may indicate performance issues with the server or the components interacting with it.
+ 
+High latencies may be caused by a number of different factors, such as inadequate hardware resources, inefficient queries, or contention between concurrent operations. By monitoring this metric, it is possible to quickly identify any potential performance issues and take appropriate corrective action. Monitoring this metric can help to avoid any unexpected downtime or performance degradation.
+
+![image](https://user-images.githubusercontent.com/24860547/212876030-46a49c5a-8d7b-4cc2-b040-60638cdd043a.png)
+
+### Connections
+ - **Connections Count**: The total number of connections to the MongoDB server. Current connections and available connections are displayed as separate dimensions. 
+
+![image](https://user-images.githubusercontent.com/24860547/212876099-6ceed111-9663-4cb6-89db-76d3c33bcf25.png)
+
+ - **Connections by Rate**: The rate of connections created per second to the MongoDB server. 
+
+This metric provides insight into the traffic that is coming in to the MongoDB server. This will help identify spikes in traffic that could be caused by malicious activity or a sudden surge in usage from legitimate users. By monitoring this metric, you can identify potential bottlenecks or issues before they become problematic. Additionally, it can help you identify when you need to scale your MongoDB server to accommodate additional traffic. By monitoring this metric, you can prevent slowdowns and other issues related to an overloaded server.
+
+ - **Connections by State**: The distribution of connection states for the MongoDB server. The following states are displayed as separate dimensions: active, threaded, exhaustIsMaster, exhaustHello, awaiting_topology_changes. 
```

Comment:
Chart snapshot missing?

Reply:
Got my screenshots from Ralph's space - but it doesn't have replication and some other stuff enabled. Some of the charts like this one just showed empty so I didn't get them.